tests/std/tests/GH_002030_asan_annotate_string/test.cpp

```diff
@@ -1845,6 +1835,13 @@ void test_DevCom_10116361() {
     s1.~string();
 }
 
+void test_DevCom_10109507() {
+    // replace failed to correctly munge asan annotations while working
+    string s("abcd");
+    s.replace(0, 1, "ef", 2);
+    s.replace(0, 0, "xy", 2);

```

Comment:
Should we assert the value of `s` after each `replace` call, to minimally verify that changes are happening?